lib/Doctrine/ORM/Mapping/Exporter/FieldMetadataExporter.php

```diff
@@ -4,13 +4,14 @@
 
 namespace Doctrine\ORM\Mapping\Exporter;
 
+use Doctrine\ORM\Mapping\ColumnMetadata;
 use Doctrine\ORM\Mapping\FieldMetadata;
 
 class FieldMetadataExporter extends LocalColumnMetadataExporter
 {
     public const VARIABLE = '$property';
 
-    protected function exportInstantiation(FieldMetadata $metadata) : string
+    protected function exportInstantiation(ColumnMetadata $metadata) : string
```

Comment:
We should keep contravariance information either in `@param` or in `assert(... instanceof ...)` to help static analysis in future - i.e. `$metadata->getName()` would be undefined on ColumnMetadata.

Reply:
I've added asserts, as I think that makes the most sense.